src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure.Core/CommandBuilders/ASPNetBenchmarks.CommandBuilder.cs

```diff
@@ -72,14 +72,45 @@ public static (string, string) Build(ASPNetBenchmarksConfiguration configuration
                 commandStringBuilder.Append($" --application.options.traceOutput {Path.Combine(configuration.Output.Path, run.Key, (baseConfiguration.Key + "." + collectType)) + traceFileSuffix}");
             }
 
-            commandStringBuilder.Append($" --application.framework net8.0 ");
+            // Add any additional arguments specified.
+            if (!string.IsNullOrEmpty(configuration.benchmark_settings.additional_arguments))
+            {
+                commandStringBuilder.Append($" {configuration.benchmark_settings.additional_arguments} ");
+            }
+
+            // Add the Framework version.
+            string frameworkVersion = "net8.0";
+            // If the framework version at the top level is explicitly stated, use it.
+            if (!string.IsNullOrEmpty(configuration.Environment.framework_version))
+            {
+                frameworkVersion = configuration.Environment.framework_version;
 
-            string corerunToSend = run.Value.corerun.EndsWith("\\") ? run.Value.corerun.Remove(run.Value.corerun.Length - 1) : run.Value.corerun;
-            commandStringBuilder.Append($" --application.options.outputFiles {Path.Combine(Path.GetDirectoryName(corerunToSend), "*.*" )}");
+                // If the framework version is set at the run level, use it.
+                if (!string.IsNullOrEmpty(run.Value.framework_version))
+                {
+                    frameworkVersion = run.Value.framework_version;
+                }
+            }
```

Comment:
nit/style comment:

If you have/define a helper method somewhere

``` c#
    public static string NullIfEmpty(this string value) => String.IsNullOrEmpty(value) ? null : value;
```

then this could be
``` c#
// Choose in preference order
string frameworkVersion = configuration.Environment.framework_version.NullIfEmpty()
    ?? run.Value.framework_version.NullIfEmpty()
    ?? "net8.0";
```
(also consider if you want to use IsNullOrWhiteSpace instead, though that seems far less common than somehow ending up with the empty string)

Reply:
I read the original code backwards.  The order is flipped and should be

``` c#
// Choose in preference order
string frameworkVersion = run.Value.framework_version.NullIfEmpty()
    ?? configuration.Environment.framework_version.NullIfEmpty()
    ?? "net8.0";
```

(but also if my other comment is correct, the "net8.0" will never be used here because the second value will be "net8.0" if not overridden to be something else - unless it is possible to override it with empty/null)

Reply:
Not needed anymore since ``configuration.Environment.framework_version`` is never null.